lib/circuitry/subscriber.rb

```diff
@@ -84,6 +84,11 @@ def lock=(value)
       @lock = value
     end
 
+    def change_message_visibility(message)
+      logger.info("Retrying message now by making the 'visiblity_timeout' zero seconds for message #{message.id}")
+      sqs.change_message_visibility(queue_url: queue, receipt_handle: message.receipt_handle, visibility_timeout: 0)
+    end
+
```

Comment:
Are we sure there isn't some built in way to return / raise an exception that does this automagically... ie.

```
def run
  circuit.subscribe do |msg|
    # possibly something like this
    raise Circuit::Renequeue.new

    # or possibly something like
    return :reenqueue

    # or some other bit to signal circuitry to re-enqueue immediatly?
end
```

I feel like maybe this was private for a reason??? maybe not, it is ruby after all

Reply:
@diclophis We added this method to circuitry; so I am comfortable making it public. Come to think of it, we are using this method as such:

```
rescue => e
      change_message_visibility(message) if ignore_visibility_timeout
...
```

and we are setting: `ignore_visibility_timeout`.

What this means is we could have changed this flow to:
1. Raise in our primary application (bigmaven)
2. Circuitry would catch that exception and re-queue the message.

I am fine with what we are doing:
1. Requeueing the message
2. Exiting without raising an exception (and thus no errors in our logs etc.)

Reply:
`msg.retry!` type thng?